tests/Go/PhpUnit/JoinPointExistsConstraint.php

```diff
@@ -0,0 +1,95 @@
+<?php
+/*
+ * Go! AOP framework
+ *
+ * @copyright Copyright 2011, Lisachenko Alexander <lisachenko.it@gmail.com>
+ *
+ * This source file is subject to the license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Go\PhpUnit;
+
+use Go\TestUtils\JoinPointsExtractor;
+use \PHPUnit_Framework_Constraint as Constraint;
+use \ReflectionClass;
+use Go\Instrument\PathResolver;
+
+/**
+ * Asserts that join point exists for given class.
+ */
+class JoinPointExistsConstraint extends Constraint
+{
+    /**
+     * @var array
+     */
+    private $configuration;
+
+    public function __construct(array $configuration)
+    {
+        parent::__construct();
+        $this->configuration = $configuration;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function matches($other)
+    {
+        if (!$other instanceof JoinPoint) {
+            throw new \InvalidArgumentException(sprintf('Expected instance of "%s", got "%s".', JoinPoint::class, is_object($other) ? get_class($other) : gettype($other)));
+        }
+
+        $joinPoints = JoinPointsExtractor::extractJoinPoints($this->getPathToProxy($other->getClass()));
+
+        $access = $other->isStatic() ? 'static' : 'method';
```

Comment:
Ok, it's absolutely ok if you want to make it in another PR, just make some note for me. I'm checking whole code during review and don't mind if work in progress.